flaky test HTTPClientTests.testContentLengthTooLongFails

```
Test Case 'HTTPClientTests.testContentLengthTooLongFails' started at 2020-06-16 19:10:55.784
/code/Tests/AsyncHTTPClientTests/HTTPClientTests.swift:2323: error: HTTPClientTests.testContentLengthTooLongFails : XCTAssertThrowsError failed: did not throw error - 
/code/Tests/AsyncHTTPClientTests/HTTPClientTests.swift:2338: error: HTTPClientTests.testContentLengthTooLongFails : failed - Body missing: Response(host: "localhost", status: NIOHTTP1.HTTPResponseStatus.badRequest, version: HTTP/1.1, headers: [("Connection", "close"), ("Content-Length", "0")], body: nil)
Test Case 'HTTPClientTests.testContentLengthTooLongFails' failed (0.046 seconds)
```

--- a/Sources/AsyncHTTPClient/HTTPHandler.swift
+++ b/Sources/AsyncHTTPClient/HTTPHandler.swift
@@ -867,15 +867,17 @@ extension TaskHandler: ChannelDuplexHandler {
             return context.eventLoop.makeSucceededFuture(())
         }
 
+        let channel = context.channel
+
         func doIt() -> EventLoopFuture<Void> {
             return body.stream(HTTPClient.Body.StreamWriter { part in
                 let promise = self.task.eventLoop.makePromise(of: Void.self)
                 // All writes have to be switched to the channel EL if channel and task ELs differ
-                if context.eventLoop.inEventLoop {
+                if channel.eventLoop.inEventLoop {
                     self.actualBodyLength += part.readableBytes
                     context.writeAndFlush(self.wrapOutboundOut(.body(part)), promise: promise)
                 } else {
-                    context.eventLoop.execute {
+                    channel.eventLoop.execute {
                         self.actualBodyLength += part.readableBytes
                         context.writeAndFlush(self.wrapOutboundOut(.body(part)), promise: promise)
                     }

--- a/Tests/AsyncHTTPClientTests/HTTPClientTests.swift
+++ b/Tests/AsyncHTTPClientTests/HTTPClientTests.swift
@@ -2319,7 +2319,7 @@ class HTTPClientTests: XCTestCase {
     }
 
     func testContentLengthTooLongFails() throws {
-        let url = self.defaultHTTPBinURLPrefix + "/post"
+        let url = self.defaultHTTPBinURLPrefix + "post"
         XCTAssertThrowsError(
             try self.defaultClient.execute(request:
                 Request(url: url,
@@ -2348,7 +2348,7 @@ class HTTPClientTests: XCTestCase {
 
     // currently gets stuck because of #250 the server just never replies
     func testContentLengthTooShortFails() throws {
-        let url = self.defaultHTTPBinURLPrefix + "/post"
+        let url = self.defaultHTTPBinURLPrefix + "post"
         let tooLong = "XBAD BAD BAD NOT HTTP/1.1\r\n\r\n"
         XCTAssertThrowsError(
             try self.defaultClient.execute(request:

